### PR TITLE
Add deployment job to CI

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

**Comment by webmaster128** [comment]
Blocked by https://github.com/CosmWasm/cosmwasm-plus/issues/30

**Comment by ethanfrey** [comment]
Also blocked by https://github.com/CosmWasm/rust-optimizer/issues/18

**Comment by ethanfrey** [comment]
I closed #30 and https://github.com/CosmWasm/rust-optimizer/pull/19 is ready for review (to fix the build errors)

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,15 +1,23 @@
 version: 2
 workflows:
   version: 2
-  test:
+  # test:
```
Comment (ethanfrey) [inline .circleci/config.yml:4]: Why are these all commented out?
I assume just as WIP while testing the other script, right?

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-      - contract_cw1_whitelist
-      - contract_cw20_base
-      - contract_cw20_escrow
-      - package_cw1
-      - package_cw20
-      - lint
+      - build_and_upload_contracts
+          # filters:
+          #   tags:
+          #     only: /^v[0-9]+\.[0-9]+\.[0-9]+.*/
```
Comment (ethanfrey) [inline .circleci/config.yml:18]: Yeah, and other code to enable once it works. Got it.

File: .circleci/config.yml [hunk 3]
```diff
@@ -292,3 +300,52 @@ jobs:
...
+          command: |
+            # create a dummy container which will hold a volume with config
+            docker create -v /code --name with_code alpine /bin/true
+            # copy a config file into this volume
+            docker cp ./contracts with_code:/code
+            docker cp ./packages with_code:/code
+      - run:
+          name: Build development contracts
+          command: |
+            for contract_dir in ./contracts/*/; do
```
Comment (ethanfrey) [inline .circleci/config.yml:330]: Okay, this needs to be updated for 0.10.0 rust-optimizer, but the rest looks quite straight-forward.

Thanks for this PR

**Comment by ethanfrey** [comment]
I updated this to rust-optimizer:0.10.0. Let's see if it works now

**Comment by ethanfrey** [comment]
Note cw1-whitelist is super small, this is due to the fact the "library" feature is activated from the cw1-subkeys import, which affects the normal `cw1_whitelist.wasm` build as well. Since we don't use this contract, I can live with it, but time to revisit the build process a bit

File: .circleci/config.yml (hunk 1 above)
Comment (webmaster128) [reply .circleci/config.yml:4]: Yes, in order to test the new job. You can revert or remove the commit "Dactivate test workflow" later

File: .circleci/config.yml (hunk 2 above)
Comment (webmaster128) [reply .circleci/config.yml:18]: Yes, to activate the filter, you need to add a `:` after `build_and_upload_contracts` as well

File: .circleci/config.yml [hunk 4]
```diff
@@ -292,3 +300,51 @@ jobs:
...
+            docker cp Cargo.toml with_code:/code
+            docker cp Cargo.lock with_code:/code
+            docker cp ./contracts with_code:/code
+            docker cp ./packages with_code:/code
+      - run:
+          name: Build development contracts
+          command: |
+            docker run --volumes-from with_code cosmwasm/rust-optimizer:0.10.0
+            docker cp with_code:/code/artifacts ./artifacts
+            mv artifacts/contracts.txt artifacts/checksums.txt
```
Comment (webmaster128) [inline .circleci/config.yml:331]: If we do that, we better call the file `checksums.txt` everywhere. I used that name in ComsJS but we can switch it to `contracts.txt`. However, I think `checksums.txt` is a bit more user friendly.

Comment (ethanfrey) [reply .circleci/config.yml:331]: Okay, I will update rust-optimizer to 0.10.1 with that naming. Then I can merge this, right?

Comment (webmaster128) [reply .circleci/config.yml:331]: Yeah, looks good to me. Just the deactivated publishing is missing but this you will figure out by bruteforceing it

Comment (ethanfrey) [reply .circleci/config.yml:331]: Addressed by https://github.com/CosmWasm/rust-optimizer/pull/20

Comment (webmaster128) [reply .circleci/config.yml:331]: Perfect. We'll benefit from this consistency in the long run

**Comment by ethanfrey** [comment]
Oops, I need to add a github token here. I will add it and try re-running the build

**Pull request merged**